app/uk/gov/hmrc/helptosavestub/controllers/EligibilityCheckController.scala

```diff
@@ -42,32 +43,55 @@ class EligibilityCheckController extends BaseController {
     8 → "Entitled to WTC and in receipt of positive WTC/CTC Tax Credit and in receipt of UC and income sufficient"
   )
 
-  val eligibleReasonGen: Gen[EligibilityCheckResult] =
-    Gen.choose(6, 8) // scalastyle:ignore magic.number
-      .map{ i ⇒
-        val reason = reasonMappings.getOrElse(i, sys.error(s"Could not find eligibility reason for code $i"))
-        EligibilityCheckResult("Eligible to HtS Account", 1, reason, i)
-      }
-
-  val ineligibleReasonGen: Gen[EligibilityCheckResult] =
-    Gen.choose(2, 5) // scalastyle:ignore magic.number
-      .map{ i ⇒
-        val reason = reasonMappings.getOrElse(i, sys.error(s"Could not find ineligibility reason for code $i"))
-        EligibilityCheckResult("Ineligible to HtS Account", 2, reason, i)
-      }
-
   val alreadyHasAccountResult: EligibilityCheckResult =
     EligibilityCheckResult("HtS account already exists", 3, "HtS account was previously created", 1)
 
+  val invalidResultCode: EligibilityCheckResult =
+    EligibilityCheckResult("INVALID RESULT WHICH DES SHOULD NEVER SEND", 4, "Not entitled to WTC and not in receipt of UC", 2)
+
+  def eligibleResult(reasonCode: Int): EligibilityCheckResult = {
+    val reason = reasonMappings.getOrElse(reasonCode, sys.error(s"Could not find eligibility reason for code $reasonCode"))
+    EligibilityCheckResult("Eligible to HtS Account", 1, reason, reasonCode)
+  }
+
+  def ineligibleResult(reasonCode: Int): EligibilityCheckResult = {
+    val reason = reasonMappings.getOrElse(reasonCode, sys.error(s"Could not find eligibility reason for code $reasonCode"))
+    EligibilityCheckResult("Ineligible to HtS Account", 2, reason, reasonCode)
+  }
+
+  def getReasonCodeFromNino(nino: String): Int =
+    Try(nino.substring(3, 4).toInt)
+      .getOrElse(sys.error(s"Error getting reason code from fourth character of NINO $nino"))
```

Comment:
does `nino.substring(3, 4)` this gives `fourth character of NINO ` really ?  or it gives fourth and fifth characters too ?

Reply:
Just the fourth character.